src/snowflake/snowpark/file_operation.py

```diff
@@ -239,22 +239,25 @@ def put_stream(
         Returns:
             An object of :class:`PutResult` which represents the results of an uploaded file.
         """
+        stage_location = stage_location.strip()
+        if not stage_location:
+            raise ValueError("stage_location cannot be empty")
+        if stage_location[-1] == "/":
+            raise ValueError("stage_location should end with target filename")
+
         if is_in_stored_procedure():
-            raise NotImplementedError(
-                "Stream uploads for stored procedure are not supported yet"
-            )
+            options = {
+                "parallel": parallel,
+                "source_compression": source_compression,
+                "auto_compress": auto_compress,
+                "overwrite": overwrite,
```

Comment:
Do we have test code to cover all these options?

Reply:
pretty much. `overwrite` and `auto_compress` are the main ones. I also added a test to see if auto detection for source compression works with gzip.